demo/index.html

```diff
@@ -0,0 +1,147 @@
+<!doctype html>
+<html lang="en-US">
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Book Visual Adjustments</title>
+  <style>
+    @media (prefers-color-scheme: dark) {
+      body {
+        color: #ffffff;
+        background: #181818;
+      }
+    }
+    :root {
+      --menuButtonLabelDisplay: block;
+      --menuSliderBg: #151515;
+      --primaryTextColor: #fff;
+      --menuWidth: 320px;
+      --animationTiming: .25s;
+      --iconFillColor: #fff;
+      --iconStrokeColor: #fff;
+      --iconWidth: 28px;
+      --iconHeight: 28px;
+      --activeButtonBg: #282828;
+    }
+
+    html {
+      font: normal 10px "Helvetica Neue", Helvetica, Arial, sans-serif;
+    }
+
+    body {
+      margin: 0;
+    }
+
+    button {
+      height: 2rem;
+      margin: 1rem;
+      grid-column: 1 / 2;
+    }
+
+    main {
+      display: grid;
+      height: 100vh;
+      grid-template-columns: auto 1fr;
+      grid-template-rows: 1fr auto;
+    }
+
+    #menu {
+      position: relative;
+      grid-column: 1 / 2;
+      width: var(--menuWidth);
+    }
+
+    picture {
+      display: block;
+      grid-column: 2 / 3;
+      grid-row: 1 / 3;
+    }
+
+    img {
+      display: block;
+      margin: 0 auto;
+    }
+  </style>
+</head>
+<body>
+  <button>Toggle menu</button>
+  <main>
+    <div id="menu"></div>
+    <picture>
+      <img src="https://ia801302.us.archive.org/BookReader/BookReaderImages.php?zip=/29/items/goody/goody_jp2.zip&file=goody_jp2/goody_0074.jp2&id=goody&scale=4&rotate=0" />
+    </picture>
+  </main>
+
+  <script type="module">
+    import { render } from 'lit-html';
+    import { html, css, LitElement } from 'lit-element';
+    import '@internetarchive/icon-visual-adjustment/icon-visual-adjustment';
+    import '@internetarchive/ia-menu-slider/ia-menu-slider';
+    import '../ia-book-visual-adjustments.js';
+
+    const menuSlider = document.createElement('ia-menu-slider');
+    const visualAdjustmentOptions = [{
+      id: 'grayscale',
+      name: 'Grayscale',
+      active: false,
+    }, {
+      id: 'brightness',
+      name: 'Adjust brightness',
+      active: false,
+      min: 0,
+      max: 150,
+      step: 1,
+      value: 100,
```

Comment:
How does the system know whether to show a checkbox or a slider?

Reply:
If value attribute is present, the range is needed to express this value instead of a simple checkbox. Felt redundant to add both a value and a `range: true`.

Reply:
Hrm yeah, but I think it'd also be beneficial to be explicit about the type instead of inferring it from other parameters. ie passing in `type: checkbox` or `type: range`.